src/netlink.c

```diff
@@ -559,9 +581,9 @@ static int qos_filter_cb (const struct nlmsghdr *nlh, void *args)
       return MNL_CB_ERROR;
     }
 
-    mnl_attr_parse_nested (attr, qos_attr_cb, &bs);
+    mnl_attr_parse_nested (attr, qos_attr_cb, &q_stats);
 
-    if (bs != NULL)
+    if (bs != NULL || qs != NULL)
```

Comment:
Both, `bs` and `qs` will *always* be `NULL`. You need to check `q_stats.bs` and `q_stats.qs` here. The `bs` and `qs` variables are unused and can be deleted.